### PR TITLE
feat: Speaker/Session CSV/PDF Export should adhere to filters

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes https://github.com/fossasia/open-event-frontend/issues/6773

Speaker/Session CSV/PDF Export should adhere to filters

required for https://github.com/fossasia/open-event-frontend/pull/7343

### CODE REVIEW

File: app/api/helpers/tasks.py [hunk 1]
```diff
@@ -565,8 +565,29 @@ def export_attendees_pdf_task(self, event_id):
...
+        'confirmed',
+        'rejected',
+        'withdrawn',
+        'canceled',
+    ]:
+        status = 'all'
+
+    if status == 'all':
+        sessions = Session.query.filter(
+            Session.event_id == event_id, Session.deleted_at == None
```
Comment (ranjitkshah) [inline app/api/helpers/tasks.py:582]:
```suggestion
            Session.event_id == event_id, Session.deleted_at is None
```

File: app/api/helpers/tasks.py [hunk 2]
```diff
@@ -565,8 +565,29 @@ def export_attendees_pdf_task(self, event_id):
...
+        ).all()
+    else:
+        sessions = Session.query.filter(
+            Session.state == status,
+            Session.event_id == event_id,
+            Session.deleted_at == None,
```
Comment (ranjitkshah) [inline app/api/helpers/tasks.py:588]:
```suggestion
            Session.deleted_at is None,
```

File: app/api/helpers/tasks.py [hunk 3]
```diff
@@ -595,8 +616,37 @@ def export_sessions_csv_task(self, event_id):
...
+        'rejected',
+        'withdrawn',
+        'canceled',
+        'without_session',
+    ]:
+        status = 'all'
+
+    if status == 'without_session':
+        speakers = Speaker.query.filter(
+            Speaker.sessions == None,
```
Comment (ranjitkshah) [inline app/api/helpers/tasks.py:635]:
```suggestion
            Speaker.sessions is None,
```

File: app/api/helpers/tasks.py [hunk 4]
```diff
@@ -595,8 +616,37 @@ def export_sessions_csv_task(self, event_id):
...
+            Speaker.event_id == event_id,
+            Speaker.deleted_at == None,
```
Comment (ranjitkshah) [inline app/api/helpers/tasks.py:637]:
```suggestion
            Speaker.deleted_at is None,
```

File: app/api/helpers/tasks.py [hunk 5]
```diff
@@ -595,8 +616,37 @@ def export_sessions_csv_task(self, event_id):
...
+        ).all()
+    elif status == 'all':
+        speakers = Speaker.query.filter(
+            Speaker.event_id == event_id, Speaker.deleted_at == None
```
Comment (ranjitkshah) [inline app/api/helpers/tasks.py:641]:
```suggestion
            Speaker.event_id == event_id, Speaker.deleted_at is None
```

File: app/api/helpers/tasks.py [hunk 6]
```diff
@@ -595,8 +616,37 @@ def export_sessions_csv_task(self, event_id):
...
+        ).all()
+    else:
+        speakers = Speaker.query.filter(
+            Speaker.sessions.any(state=status),
+            Speaker.event_id == event_id,
+            Speaker.deleted_at == None,
```
Comment (ranjitkshah) [inline app/api/helpers/tasks.py:647]:
```suggestion
            Speaker.deleted_at is None,
```

**Review by ranjitkshah — CHANGES_REQUESTED** [review]

**Comment by ranjitkshah** [comment]
lgtm fix left! , 
 works!!

**Review by ranjitkshah — APPROVED** [review]

**Review by MeghalBisht — APPROVED** [review]

**Pull request merged**